### PR TITLE
Refactor Civo Kubernetes node pool configuration and organize code in their own folders per service

### PR DESCRIPTION
**Pull request by alejandrojnm**
This pull request refactors the Civo Kubernetes node pool configuration and organizes the code in their own folders per service. It also includes updates to the following files:

- civo_kubernetes_cluster.tf
- disk/data_source_disk_image.go
- database/resource_database.go
- dns/resource_dns_domain_name.go
- dns/resource_dns_domain_record.go
- volume/resource_volume_attachment.go
- network/resource_network.go
- firewall/resource_firewall.go
- kubernetes/data_source_kubernetes_cluster.go
- network/data_source_network.go
- volume/resource_volume.go
- firewall/resource_firewall.go
- ip/data_source_reserved_ip.go
- ssh/data_source_ssh_key.go
- dns/data_source_dns_domain_name.go
- ip/resource_reserved_ip.go
- size/data_source_size.go
- dns/data_source_dns_domain_record.go
- instances/resource_instance_reserved_ip_assignment.go
- database/data_source_database.go

Signed-off-by: Alejandro J. Nuñez Madrazo <alejandrojnm@gmail.com>

### CODE REVIEW

File: civo/database/resource_database_test.go [hunk 1]
```diff
@@ -44,33 +46,34 @@ func TestAccCivoDatabase_basic(t *testing.T) {
 	})
 }
 
-func TestAccCivoDatabase_update(t *testing.T) {
+// CivoDatabase_update is used to test the database resource
+func CivoDatabase_update(t *testing.T) {
```
Comment (alessandroargentieri) [inline civo/database/resource_database_test.go:50]: Why not calling `Test<FunctionUnderTest>` the test?

File: civo/volumen/resource_volume_attachment.go
Comment (alessandroargentieri) [inline civo/volumen/resource_volume_attachment.go:1]: rename `volumen` into `volume`

**Review by alessandroargentieri — CHANGES_REQUESTED** [review]
For now, a test (a function accepting `t *testing.T` should be prefixed with `Test<FunctionUnderTest>`, wondering why you removed this convention

File: civo/database/resource_database_test.go (hunk 1 above)
Comment (alejandrojnm) [reply civo/database/resource_database_test.go:50]: I will check

File: civo/volumen/resource_volume_attachment.go
Comment (alejandrojnm) [reply civo/volumen/resource_volume_attachment.go:1]: True

File: civo/acceptance/acceptance.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	}
+	TestAccProviderFactories = map[string]func() (*schema.Provider, error){
+		"civo": func() (*schema.Provider, error) {
+			return TestAccProvider, nil
+		},
+	}
+}
+
+// TestProvider - Test the provider itself
+func TestProvider(t *testing.T) {
```
Comment (alessandroargentieri) [inline civo/acceptance/acceptance.go:41]: the tests should be inserted into `<namefile>_test.go` file, else are not executed.

**Review by alessandroargentieri — CHANGES_REQUESTED** [review]
Seen some tests in a non-test file. The remaining things seem good!

File: civo/acceptance/acceptance.go (hunk 2 above)
Comment (alejandrojnm) [reply civo/acceptance/acceptance.go:41]: No, this one is fine, it is just and utility

Comment (alessandroargentieri) [reply civo/acceptance/acceptance.go:43]: IMO test utilities should return objects and function for tests, not having tests inside. I trust your expertise.

**Review by alessandroargentieri — APPROVED** [review]
LGTM

**Pull request merged**